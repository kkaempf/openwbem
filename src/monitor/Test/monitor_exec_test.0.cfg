unpriv_user { owprovdr }

read_dir {
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/
  /dir/doesnt/exist/
}

rename_from {
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename1
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename2
  /dir/doesnt/exist/rename
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/no-such-file
}

rename_to {
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to
  /dir/doesnt/exist/rename-to
}

unlink {
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/unlink-exists
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/unlink-doesnt-exist
  /dir/doesnt/exist/unlink
}

open_r {
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/bar
}

open_rw {
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-*
}

open_w {
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/out-*
}

open_a {
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/app-*
}
