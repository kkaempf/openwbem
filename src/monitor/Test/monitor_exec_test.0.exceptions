PrivilegeManagerException   CheckException: PrivilegeManager::readDirectory: could not read /dir/doesnt/exist/
PrivilegeManagerException   CheckException: PrivilegeManager::rename: could not rename /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename2 to /dir/doesnt/exist/rename-to
PrivilegeManagerException   CheckException: PrivilegeManager::rename: could not rename /dir/doesnt/exist/rename to /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to
PrivilegeManagerException   CheckException: PrivilegeManager::rename: could not rename /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/no-such-file to /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to
PrivilegeManagerException   CheckException: PrivilegeManager::open:
