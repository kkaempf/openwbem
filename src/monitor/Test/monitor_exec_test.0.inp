# These test cases where there are no protocol errors

readDirectory /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/ keep_special
readDirectory /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/ omit_special
readDirectory /dir/doesnt/exist/ keep_special
readDirectory /dir/doesnt/exist/ omit_special

rename /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename1 /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to
rename /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename2 /dir/doesnt/exist/rename-to
rename /dir/doesnt/exist/rename /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to
rename /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/no-such-file /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to

unlink /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/unlink-exists
unlink /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/unlink-doesnt-exist
unlink /dir/doesnt/exist/unlink

open in 0 /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/bar
open out 604 /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/out-exists
  piffle-prog
open out 606 /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/out-create
  hack-phlegm
open out|trunc 604
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/out-trunc-exists
  enigmatic
open out|trunc 604
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/out-trunc-create
  wizard-of-oz
open out|app 700
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/app-exists
  send-in-the-clowns
open out|app 500 
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/app-create
  horatio-at-the-bridge
open in|out 404 
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-exists
  xxx
open in|out 400 
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-nonesuch
  foo-bar-baz
open in|out|trunc 606 
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-trunc-exists
  yyy
open in|out|trunc 606 
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-trunc-create
  This-is-the-song-that-never-ends
open in|out|ate 402
  /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-exists-ate
  xxx
