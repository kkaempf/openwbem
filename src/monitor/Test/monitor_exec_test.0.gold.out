argv:
  montest;
  arg1;
  arg2;
environ:
  XPATH=XXX;
  IFS= ;
  XPATH=XXX;
  XPATH=XXX;
  OW_PRIVILEGE_MONITOR_DESCRIPTOR=XXX;
  PATH=/bin;
  XPATH=XXX;
  foo=bar;
User: owprovdr

readDirectory
  .
  ..
  app-exists
  bar
  foo
  in-out-exists
  in-out-exists-ate
  in-out-trunc-exists
  out-exists
  out-trunc-exists
  rename1
  rename2
  unlink-exists

readDirectory
  app-exists
  bar
  foo
  in-out-exists
  in-out-exists-ate
  in-out-trunc-exists
  out-exists
  out-trunc-exists
  rename1
  rename2
  unlink-exists

readDirectory
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::readDirectory: could not read /dir/doesnt/exist/

readDirectory
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::readDirectory: could not read /dir/doesnt/exist/

rename

rename
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::rename: could not rename /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename2 to /dir/doesnt/exist/rename-to

rename
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::rename: could not rename /dir/doesnt/exist/rename to /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to

rename
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::rename: could not rename /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/no-such-file to /monexectest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to

unlink
1

unlink
0

unlink
0

open
  Hokey smokes, Bullwinkle!


open

open

open

open

open

open

open
  xxxtle Bo Peep has lost her sheep


open
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::open: 2(No such file or directory)

open
  yyy

open
  This-is-the-song-that-never-ends

open
  Little Bo Peep has lost her sheep
xxx
ERROR OUTPUT: err output
